nncf/openvino/graph/model_transformer.py

```diff
@@ -235,24 +235,21 @@ def _apply_quantizer_insertion_transformations(
         return model
 
     @staticmethod
-    def convert_params_to_fp16(
-        fq_params: FakeQuantizeParameters,
-    ) -> Tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
+    def convert_params_to_operations(
+        fq_params: FakeQuantizeParameters, dtype: ov.Type
+    ) -> Tuple[ov.Node, ov.Node, ov.Node, ov.Node]:
         """
-        Converts FakeQuantize parameters to FP16 precision.
+        Converts FakeQuantize parameters to operations with provided dtype.
 
         :param fq_params: FakeQuantize node attributes.
-        :return: FakeQuantize parameters in FP16 precision.
+        :param dtype: Data type for operations.
+        :return: FakeQuantize parameters as operations.
         """
 
-        def _convert_to_fp16(data):
-            clip_data = np.clip(data, np.finfo(np.float16).min, np.finfo(np.float16).max)
-            return clip_data.astype(np.float16)
-
-        input_low = _convert_to_fp16(fq_params.input_low.data)
-        input_high = _convert_to_fp16(fq_params.input_high.data)
-        output_low = _convert_to_fp16(fq_params.output_low.data)
-        output_high = _convert_to_fp16(fq_params.output_high.data)
+        input_low = opset.constant(value=fq_params.input_low.data, dtype=dtype)
+        input_high = opset.constant(value=fq_params.input_high.data, dtype=dtype)
+        output_low = opset.constant(value=fq_params.output_low.data, dtype=dtype)
+        output_high = opset.constant(value=fq_params.output_high.data, dtype=dtype)
```

Comment:
Are you sure that we don't need clip FQ params before creating the constant node?

Reply:
Good catch. I'll look at this.